client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -261,6 +261,35 @@ public synchronized int getId(String subject, Schema schema)
     }
   }
 
+  @Override
+  public List<Integer> deleteSubject(String subject) throws IOException, RestClientException {
+    return deleteSubject(restService.DEFAULT_REQUEST_PROPERTIES, subject);
+  }
+
+  @Override
+  public List<Integer> deleteSubject(Map<String, String> requestProperties, String subject)
+      throws IOException, RestClientException {
+    versionCache.remove(subject);
+    idCache.remove(subject);
+    schemaCache.remove(subject);
+    return restService.deleteSubject(requestProperties, subject);
+  }
+
+  @Override
+  public Integer deleteSchemaVersion(String subject, String version)
+      throws IOException, RestClientException {
+    return deleteSchemaVersion(RestService.DEFAULT_REQUEST_PROPERTIES, subject, version);
+  }
+
+  @Override
+  public Integer deleteSchemaVersion(
+      Map<String, String> requestProperties,
+      String subject,
+      String version)
+      throws IOException, RestClientException {
+    return restService.deleteSchemaVersion(requestProperties, subject, version);
```

Comment:
missing the cache delete in this method. The tests should simply assert that you can get the schema by id but not by version